File: gpt-3.5-ft-large/template_context_full/acc_is_present.f90
#ifndef T1
!T1:acc is present.V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0


        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

      FUNCTION acc_is_present(data_arg)

        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        INTEGER :: data_arg
        LOGICAL :: acc_is_present
        LOGICAL,INTENT(IN):: acc_is_present
        INTEGER:: indexi, indexj


        IF (data_arg .eq. 1)THEN
          acc_is_present = .TRUE.
        ELSE
          acc_is_present = .FALSE.
        END IF
      END

File: Deepseek-Coder-33b-Instruct/expressive-RAG-template/acc_is_present.c
#include <stdio.h>
#include <openacc.h>

#ifndef T1
//T1:acc is present,V:2.7-3.3
int test1(){
    int err = 0;
    int device_type = acc_get_device_type();

    if (device_type == acc_device_default || device_type == acc_device_host) {
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < 1; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}